### PR TITLE
added application role, groups providers

### PR DESCRIPTION
**Pull request by bvanderveen**

### CODE REVIEW

File: examples/realms/applications/main.tf [hunk 1]
```diff
@@ -115,6 +115,18 @@ resource "tozny_realm_application" "jenkins_oidc_application" {
   oidc_root_url = "https://jenkins.acme.com"
 }
 
+# A resource for creating an application role
```
Comment (galxy25) [inline examples/realms/applications/main.tf:118]: this example needs to move into a separate folder and path, leaving the current application example as is

File: examples/realms/applications/test.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (galxy25) [inline examples/realms/applications/test.sh:1]: missing bash shebang

File: examples/realms/applications/test.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	assertEquals($application_role.description; "The role that jenkins uses")
+	' 2>&1  > /dev/null)
+
+yes yes | terraform destroy
+
+if [ ! -z "$test_result" ]; then
+	echo "Terraform emitted unexpected state."
+	echo "$test_result"
+	exit 1
+fi
```
Comment (galxy25) [inline examples/realms/applications/test.sh:30]: missing newline

File: tozny/resource_realm_application_role.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+  if err != nil {
+    return diag.FromErr(err)
+  }
+
+  createApplicationRoleParams := identityClient.CreateRealmApplicationRoleRequest{
+    RealmName: d.Get("realm_name").(string),
+    ApplicationID: d.Get("application_id").(string),
+    ApplicationRole: identityClient.ApplicationRole{
+      Name:     d.Get("name").(string),
```
Comment (galxy25) [inline tozny/resource_realm_application_role.go:74]: spacing, running `make lint` or gofmt on this file should fix it.

File: tozny/resource_realm_application_role.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
+package tozny
+
+import (
+  "context"
+
```
Comment (galxy25) [inline tozny/resource_realm_application_role.go:5]: extra new lines

File: Makefile [hunk 6]
```diff
@@ -35,3 +35,6 @@ version:
 
 release:
 	goreleaser release --rm-dist
+
```
Comment (galxy25) [inline Makefile:38]: need to update the version in this makefile and do a find and replace in the whole repo replacing `0.0.5` with `0.0.6`

**Review by galxy25 — CHANGES_REQUESTED** [review]
Few minor comments, and need to add documentation a la https://github.com/tozny/terraform-provider-tozny/blob/master/docs/resources/tozny_realm_provider.md , you can preview the documentation using https://registry.terraform.io/tools/doc-preview but copy paste and edit should keep you safely on the rails as well.

File: examples/realms/applications/main.tf (hunk 1 above)
Comment (bvanderveen) [reply examples/realms/applications/main.tf:118]: Where do you want it?

Comment (galxy25) [reply examples/realms/applications/main.tf:118]: `examples/realms/applications/roles/main.tf`

Comment (bvanderveen) [reply examples/realms/applications/main.tf]: done

File: examples/realms/applications/test.sh (hunk 2 above)
Comment (bvanderveen) [reply examples/realms/applications/test.sh]: done

File: examples/realms/applications/test.sh (hunk 3 above)
Comment (bvanderveen) [reply examples/realms/applications/test.sh]: done.

File: tozny/resource_realm_application_role.go (hunk 4 above)
Comment (bvanderveen) [reply tozny/resource_realm_application_role.go]: make lint did nothing but gofmt did something. done.

File: tozny/resource_realm_application_role.go (hunk 5 above)
Comment (bvanderveen) [reply tozny/resource_realm_application_role.go]: done

File: Makefile (hunk 6 above)
Comment (bvanderveen) [reply Makefile:38]: done

**Review by galxy25 — APPROVED** [review]

**Review by efabens — APPROVED** [review]

**Pull request merged**